### PR TITLE
feat(seo): create tab for seo settings

### PR DESCRIPTION
**Pull request by lwhiteley**
**Updates:**
Add SEO settings tab
Let me know if any changes should be made

See screenshot below to see what it looks like:

seo view

**Notes:**
- I wanted to make Description a Textarea but not sure how to do that.

**Future Considerations:**
- Make each property overridable per page
- Add tab for [Social Media meta tags](https://moz.com/blog/meta-data-templates-123) support

**Read more on SEO:**
- http://www.wordstream.com/meta-tags

### CODE REVIEW

File: lib/server/index.js [hunk 1]
```diff
@@ -60,9 +62,15 @@ app.use('/graphql', graphqlHTTP(req => ({
...
+
+  const metaSettings = {
+    'description': 'description', 'keywords': 'keywords',
+    'copyright':'copyright', 'robots':'robots', 'dctitle': 'DC.title'
+  };
+
   const settingsArr = await SettingModel
     .find({
-      _id: {$in: ['title', 'favicon']}
+      _id: {$in: ['title', 'favicon'].concat(_keys(metaSettings))}
```
Comment (bruno12mota) [inline lib/server/index.js]: Use Object.keys instead. You can drop the lodash.keys dep. You can also use es6 spread:

```js
{$in: ['title', 'favicon', ...Object.keys(metaSettings)]}
```

File: lib/server/index.js [hunk 2]
```diff
@@ -60,9 +62,15 @@ app.use('/graphql', graphqlHTTP(req => ({
 })));
 
 app.use(async (req, res, next) => {
+
+  const metaSettings = {
+    'description': 'description', 'keywords': 'keywords',
+    'copyright':'copyright', 'robots':'robots', 'dctitle': 'DC.title'
+  };
```
Comment (bruno12mota) [inline lib/server/index.js]: Check eslint validation. Don't use strings as keys, and for coherence divide each item in a new line:

```js
const metaSettings = {
  description: 'description',
  keywords: 'keywords',
  copyright: 'copyright'
  robots: 'robots',
  dctitle: 'DC.title'
};
```

File: lib/server/index.js [hunk 3]
```diff
@@ -74,6 +82,18 @@ app.use(async (req, res, next) => {
     }
   ];
 
+  _forEach(metaSettings, ( (value, key) => {
+    res.locals.header.push(
```
Comment (bruno12mota) [inline lib/server/index.js]: * Check if setting has value before adding metatag, avoiding empty meta tags
* For coherence change `_forEach` to `forEach`:
* You can also remove the parentheses surrounding the cb function

```js
  forEach(metaSettings, (value, key) => {
    if (settings[key]) {
      res.locals.header.push({
        tag: 'meta',
        props:{
          name: value,
          content: safeHtmlString(settings[key])
        }
      });
    }
  });
```

File: lib/shared/screens/admin/screens/settings/screens/general/index.js [hunk 4]
```diff
@@ -45,6 +45,7 @@ const options = [
       height: 114
     }
   }
+
```
Comment (bruno12mota) [inline lib/shared/screens/admin/screens/settings/screens/general/index.js]: Unnecessary break line

File: package.json [hunk 5]
```diff
@@ -49,6 +49,7 @@
     "lodash.foreach": "^3.0.3",
     "lodash.get": "https://registry.npmjs.org/lodash.get/-/lodash.get-4.4.2.tgz",
     "lodash.intersection": "^4.2.0",
+    "lodash.keys": "^4.2.0",
```
Comment (bruno12mota) [inline package.json]: Drop lodash.keys dep

**Comment by lwhiteley** [comment]
@bruno12mota changes made as requested 👍

**Comment by bruno12mota** [comment]
Awesome mate, thanks for contributing! 😄

**Pull request merged**